lib/index.js

```diff
@@ -0,0 +1,56 @@
+// Knex.js  0.8.0
+// --------------
+//     (c) 2014 Tim Griesser
+//     Knex may be freely distributed under the MIT license.
+//     For details and documentation:
+//     http://knexjs.org
+
+'use strict';
+
+var Raw            = require('./raw')
+var warn           = require('./helpers').warn
+var Client         = require('./client')
+
+var makeClient     = require('./util/make-client')
+var makeKnex       = require('./util/make-knex')
+var assign         = require('lodash/object/assign')
+
+function Knex(config) {
+  if (typeof config === 'string') {
+    return new Knex(assign(parseUrl(config), arguments[2]))
+  }
+  var Dialect;
+  if (arguments.length === 0 || (!config.client && !config.dialect)) {
+    Dialect = makeClient(Client)
+  } else {
+    var clientName = config.client || config.dialect
+    Dialect = makeClient(require('./dialects/' + (aliases[clientName] || clientName) + '/index.js'))
```

Comment:
this will not work in browserify you'll need to make sure this gets bundled in, you can add a line that says

```js
if (process.browser) {
    require('./dialects/websql/index.js')
}
```

somewhere earlier in the page